core/src/main/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingService.scala

```diff
@@ -359,4 +552,64 @@ class SpendReportingService(
         case ex: Exception =>
           Future.failed(RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.InternalServerError, ex)))
       }
+
+  def getSpendForAllWorkspaces(
+    start: DateTime,
+    end: DateTime,
+    pageSize: Int,
+    offset: Int
+  ): Future[Option[SpendReportingResults]] =
+    traceFutureWithParent("getSpendForAllWorkspaces", ctx) { childContext =>
+      validateReportParameters(start, end)
+      for {
+        billingMap <- getBillingWithSpendPermission(childContext)
+        _ = if (billingMap.isEmpty) {
+          return Future.successful(None)
+        }
+        projectNames: Map[GoogleProjectId, WorkspaceName] = billingMap.values.flatten.toMap
+        query = getAllUserWorkspaceQuery(billingMap, pageSize, offset)
+        queryJob = setUpAllUserWorkspaceQuery(query, start, end)
+
+        job: Job <- bigQueryService.use(_.runJob(queryJob)).unsafeToFuture().map(_.waitFor())
+        _ = logSpendQueryStats(job.getStatistics[JobStatistics.QueryStatistics])
+        result = job.getQueryResults()
```

Comment:
this would be a great place to add tracing: let's get timing and visibility into how the BigQuery query is performing. Might be easiest to pull this logic out into a separate, traced, method